build.gradle

```diff
@@ -148,8 +146,8 @@ configure(opentelemetryProjects) {
         mavenLocal()
     }
 
-    sourceCompatibility = 1.7
-    targetCompatibility = 1.7
+    sourceCompatibility = 1.8
```

Comment:
Quick note: https://docs.gradle.org/current/userguide/building_java_projects.html#sec:java_cross_compilation

Reply:
we can only use the new flag if we're compiling with java9+, though, correct?

Reply:
I tried switching to the new flag and it build locally with java 14, but failed to build on CI.

Reply:
Yeah we'll want to switch to using Java 11 to build on CI and drop support for building on older to take advantage of the new features. Even 15 would be ok there are usually new compilation features (often javadoc improvements) and still works on older. We would need to change the CI to have a build and separate test jdk configured with Gradle to continue testing with Java 8.

For another issue of course :)